### PR TITLE
Feature/dynamic items per page

### PR DESCRIPTION
**Pull request by ClaudivanFilho**
#### What is the purpose of this pull request?
- Adjust the `maximum number` of items inside the Shelf based on the screen resolution
- Adjust shelf to adapt to any ExtensionPoint
- Add default value to prop `title` in the Shelf
- Change filename `theme.css` to `global.css` and import inside component
- Remove `css` reference from `pages.json`
- Remove PLACEHOLDER_PRODUCT

#### Screenshots or example usage

workspace :arrow_forward:  https://shelf-dynamic--storecomponents.myvtex.com/

#### Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Requires change to documentation, which has been updated accordingly.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,16 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
 and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html).
 
 ## [Unreleased]
+### Added
+ - **defaultProps** Add default value to prop `title` in the Shelf
+
+### Changed
+ - **file** Change filename `Theme.css` to `Global.css`
```
Comment (leticiawanderley) [inline CHANGELOG.md]: ⛏ por que o nome dos arquivos tá com letra maiúscula aqui?

File: react/ShelfContent.js [hunk 2]
```diff
@@ -98,7 +115,7 @@ class ShelfContent extends Component {
       }
       return (
         <div className={`${VTEXClasses.ITEM_EDIT_MODE} pa4`} key="1">
-          <ShelfItem extentionId="shelfitem" item={PLACEHOLDER_PRODUCT} />
+          <ShelfItem extentionId="shelfitem" />
```
Comment (leticiawanderley) [inline react/ShelfContent.js]: ⛏ "extension"

Comment (leticiawanderley) [reply react/ShelfContent.js]: Por que essa `key`?

Comment (leticiawanderley) [reply react/ShelfContent.js]: ⛏ "extensionId", "s" not "t"

File: react/ShelfContent.js [hunk 3]
```diff
@@ -98,7 +115,7 @@ class ShelfContent extends Component {
       }
       return (
         <div className={`${VTEXClasses.ITEM_EDIT_MODE} pa4`} key="1">
```
Comment (leticiawanderley) [inline react/ShelfContent.js]: Por que essa key?

File: react/ShelfContent.js [hunk 4]
```diff
@@ -8,45 +8,74 @@ import ShelfItem from './ShelfItem'
...
+      const shelfItemWidth = this.getShelfItemWidth()
+      const maxItemsPerPage = Math.floor(slideListWidth / shelfItemWidth)
+      if (this.props.itemsPerPage >= maxItemsPerPage) {
+        return maxItemsPerPage || 1
+      }
+    }
+    return this.props.itemsPerPage
+  }
+
+  configureSlideSettings(itemsLength) {
```
Comment (leticiawanderley) [inline react/ShelfContent.js]: Is this function being used?
If it isn't, remove it

File: react/ShelfContent.js [hunk 5]
```diff
@@ -8,45 +8,74 @@ import ShelfItem from './ShelfItem'
...
 /**
  * ShelfContent Component. Executes the interaction with react-slick
  * and render the properly content of the Shelf depending of edit mode state.
  */
 class ShelfContent extends Component {
-  configureSlideSettings(itemsLength) {
-    const { arrows, scroll, itemsPerPage } = this.props
+  constructor(props) {
+    super(props)
+    window.addEventListener('resize', () => {
```
Comment (klzns) [inline react/ShelfContent.js]: Adicionar o listener no `componentDidMount()`

File: react/ShelfContent.js [hunk 6]
```diff
@@ -8,45 +8,74 @@ import ShelfItem from './ShelfItem'
...
+      }
+    })
+  }
+
+  componentDidMount() {
+    this._isMounted = true
+  }
 
+  componentWillUnmount() {
+    this._isMounted = false
```
Comment (klzns) [inline react/ShelfContent.js]: Colocar `window.remodeEventListener` aqui

File: react/ShelfContent.js [hunk 7]
```diff
@@ -8,76 +8,75 @@ import ShelfItem from './ShelfItem'
...
       slidesToShow: itemsPerPage,
-      slidesToScroll: scroll === ScrollTypes.BY_PAGE.value ? itemsPerPage : 1,
-      dots: DOTS_LARGE_VIEWPORT,
-      arrows,
+      slidesToScroll: scroll === ScrollTypes.BY_PAGE.value ? itemsPerPage : MINIMUM_NUMBER_OF_ITEMS_PER_PAGE,
+      dots: hasMoreThanOneItemsPerPage && DOTS_LARGE_VIEWPORT,
+      arrows: hasMoreThanOneItemsPerPage && arrows,
       nextArrow: <Arrow cssClass={VTEXClasses.ARROW_RIGHT_CLASS} />,
       prevArrow: <Arrow cssClass={VTEXClasses.ARROW_LEFT_CLASS} />,
+      centerMode: itemsPerPage === MINIMUM_NUMBER_OF_ITEMS_PER_PAGE,
```
Comment (leticiawanderley) [inline react/ShelfContent.js]: Use `!hasMoreThanOneItemPerPage`

File: react/ShelfContent.js [hunk 8]
```diff
@@ -8,76 +8,75 @@ import ShelfItem from './ShelfItem'
...
+        return maxItemsPerPage || 1
+      }
+    }
+    return this.props.itemsPerPage
+  }
+
+  configureSlideSettings(itemsLength) {
+    const { arrows, scroll } = this.props
+    const itemsPerPage = this.getCorrectItemsPerPage() || MINIMUM_NUMBER_OF_ITEMS_PER_PAGE
+    const hasMoreThanOneItemsPerPage = itemsPerPage !== MINIMUM_NUMBER_OF_ITEMS_PER_PAGE
```
Comment (leticiawanderley) [inline react/ShelfContent.js]:
```diff
- hasMoreThanOneItemsPerPage
+ hasMoreThanOneItemPerPage
```

**Pull request merged**